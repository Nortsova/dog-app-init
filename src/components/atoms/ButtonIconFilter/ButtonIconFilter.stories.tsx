import type { Meta, StoryObj } from '@storybook/react';
import { ButtonIconFilter } from '.';

const meta = {
  title: 'Atoms/ButtonIconFilter',
  component: ButtonIconFilter,
  argTypes: {
    state: {
      options: ['default', 'down', 'random'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof ButtonIconFilter>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    state: 'default'
  }
};
