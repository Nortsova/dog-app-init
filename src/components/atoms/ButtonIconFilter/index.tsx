import { SvgIcon, styled } from '@mui/material';
import theme from '../../../theme/theme';

import IconUpFilter from '../../../assets/icons/IconUpFilter.svg?react';
import IconDownFilter from '../../../assets/icons/IconDownFilter.svg?react';
import IconRandomFilter from '../../../assets/icons/IconRandomFilter.svg?react';

type IconFilterProps = {
  state: 'default' | 'down' | 'random';
};

const BtnFilter = styled('button')({
  padding: theme.spacing(0.8, 3.5),
  borderRadius: 6,
  backgroundColor: theme.palette.grey[300],
  boxShadow: `3px 3px 0px 0px ${theme.palette.text.secondary}`,
  border: `0.6px solid ${theme.palette.common.black}`,
  transition: 'box-shadow .2s ease',
  '&:hover': {
    boxShadow: `3px 3px 0px 0px ${theme.palette.warning.main}`,
    border: `1px solid ${theme.palette.warning.main}`,
    backgroundColor: 'transparent',
    outline: 'none'
  },
  '&:active, &:focus': {
    backgroundColor: theme.palette.warning.main,
    boxShadow: `3px 3px 0px 0px ${theme.palette.text.secondary}`,
    border: `1px solid ${theme.palette.text.secondary}`
  }
});

export const ButtonIconFilter = ({ state = 'default' }: IconFilterProps) => {
  switch (state) {
    case 'down': {
      return (
        <BtnFilter>
          <SvgIcon component={IconDownFilter} inheritViewBox />
        </BtnFilter>
      );
    }
    case 'random': {
      return (
        <BtnFilter>
          <SvgIcon component={IconRandomFilter} inheritViewBox />
        </BtnFilter>
      );
    }
    default: {
      return (
        <BtnFilter>
          <SvgIcon component={IconUpFilter} inheritViewBox />
        </BtnFilter>
      );
    }
  }
};
