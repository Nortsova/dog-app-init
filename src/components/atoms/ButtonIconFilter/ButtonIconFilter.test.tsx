import { composeStories } from '@storybook/react';
import * as stories from './ButtonIconFilter.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('ButtonIconFilter', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
