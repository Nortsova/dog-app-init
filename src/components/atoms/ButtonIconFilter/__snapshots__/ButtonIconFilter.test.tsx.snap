// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIconFilter should match snapshot Component 1`] = `
.emotion-0 {
  padding: 6.4px 28px;
  border-radius: 6px;
  background-color: #E6E6E6;
  box-shadow: 3px 3px 0px 0px #000000;
  border: 0.6px solid #090808;
  -webkit-transition: box-shadow .2s ease;
  transition: box-shadow .2s ease;
}

.emotion-0:hover {
  box-shadow: 3px 3px 0px 0px #FFCF32;
  border: 1px solid #FFCF32;
  background-color: transparent;
  outline: none;
}

.emotion-0:active,
.emotion-0:focus {
  background-color: #FFCF32;
  box-shadow: 3px 3px 0px 0px #000000;
  border: 1px solid #000000;
}

<div>
  <button
    class="emotion-0"
  >
    <test-file-stub
      aria-hidden="true"
      classname="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1v8usww-MuiSvgIcon-root"
      focusable="false"
    />
  </button>
</div>
`;
